osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -152,7 +152,10 @@ public partial class ScoreProcessor : JudgementProcessor
         private readonly List<HitEvent> hitEvents = new List<HitEvent>();
         private HitObject? lastHitObject;
 
-        private double scoreMultiplier = 1;
+        /// <summary>
+        /// Should generally not be used to change multiplier. Use settter only to override default way of computing multiplier.
+        /// </summary>
+        public double ScoreMultiplier { get; set; } = 1;
```

Comment:
I'm not sure how to feel about this property being the sole hinge point of this feature but my initial thoughts are "this is not good". If you have to preface a property with "should generally not be used" it should be a big warning sign that maybe things are not entirely structured right. Anything could wipe this change and suddenly all sorts of weird stuff can happen.

If I were to provide an alternative, extensible way of making this work: have a

```csharp
public Func<IEnumerable<Mod>, double> ScoreMultiplierCalculator;
```

field. Initialise it to the default implementation of accumulating `ScoreMultiplier`. Override it to

```csharp
scoreProcessor.ScoreMultiplierCalculator = _ => 1;
```

in the pathways you need that in, and there. No more fear of something funny happening with that. (Unless something else starts messing with the calculator, but at that point, we have bigger problems.)

I guess you could also do a boolean `ApplyMultipliers` flag if needed instead.

Reply:
Good idea